Copy file panics on master

#### What is the problem you are having with rclone?

It panics.

#### What is your rclone version (output from `rclone version`)

5195075

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

Linux 64.

####  Which cloud storage system are you using? (eg Google Drive)

Local

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

```
$ ./rclone -vv copy 1GB.bin.2 /var/empty/rclone/
2019/08/09 11:59:29 DEBUG : rclone: Version "v1.48.0-145-g51950756-beta" starting with parameters ["./rclone" "-vv" "copy" "1GB.bin.2" "/var/empty/rclone/"]
2019/08/09 11:59:29 DEBUG : Using config file from "/home/michal/.config/rclone/rclone.conf"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x9a14b9]

goroutine 1 [running]:
github.com/rclone/rclone/fs/operations.moveOrCopyFile(0x1841d20, 0xc0000c8010, 0x18593c0, 0xc00058a880, 0x0, 0x0, 0x7ffd21bd8df3, 0x9, 0x7ffd21bd8df3, 0x9, ...)
        /home/michal/work/scylla/rclone/fs/operations/operations.go:1640 +0xe9
github.com/rclone/rclone/fs/operations.CopyFile(...)
        /home/michal/work/scylla/rclone/fs/operations/operations.go:1743
github.com/rclone/rclone/cmd/copy.glob..func1.1(0xc00057a860, 0xc000000180)
        /home/michal/work/scylla/rclone/cmd/copy/copy.go:81 +0xf3
github.com/rclone/rclone/cmd.Run(0xc0002c0101, 0x2368960, 0xc000541d38)
        /home/michal/work/scylla/rclone/cmd/cmd.go:234 +0xc7
github.com/rclone/rclone/cmd/copy.glob..func1(0x2368960, 0xc0002c8330, 0x2, 0x3)
        /home/michal/work/scylla/rclone/cmd/copy/copy.go:77 +0x129
github.com/spf13/cobra.(*Command).execute(0x2368960, 0xc0002c8270, 0x3, 0x3, 0x2368960, 0xc0002c8270)
        /home/michal/work/scylla/rclone/vendor/github.com/spf13/cobra/command.go:830 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0x2365760, 0x5d4d4401, 0x237c2a0, 0xc0000aa058)
        /home/michal/work/scylla/rclone/vendor/github.com/spf13/cobra/command.go:914 +0x2fc
github.com/spf13/cobra.(*Command).Execute(...)
        /home/michal/work/scylla/rclone/vendor/github.com/spf13/cobra/command.go:864
github.com/rclone/rclone/cmd.Main()
        /home/michal/work/scylla/rclone/cmd/cmd.go:499 +0x96
main.main()
        /home/michal/work/scylla/rclone/rclone.go:13 +0x20
```

--- a/fs/cache/cache.go
+++ b/fs/cache/cache.go
@@ -13,15 +13,15 @@ var (
 // GetFn gets a fs.Fs named fsString either from the cache or creates
 // it afresh with the create function
 func GetFn(fsString string, create func(fsString string) (fs.Fs, error)) (f fs.Fs, err error) {
-	value, err := c.Get(fsString, func(fsString string) (value interface{}, ok bool, error error) {
-		f, err := create(fsString)
+	value, err := c.Get(fsString, func(fsString string) (f interface{}, ok bool, err error) {
+		f, err = create(fsString)
 		ok = err == nil || err == fs.ErrorIsFile
 		return f, ok, err
 	})
-	if err != nil {
+	if err != nil && err != fs.ErrorIsFile {
 		return nil, err
 	}
-	return value.(fs.Fs), nil
+	return value.(fs.Fs), err
 }
 
 // Get gets a fs.Fs named fsString either from the cache or creates it afresh

